File: apps/pos/native-windows/DesktopKitchenPOS/Services/ReportService.cs
using DesktopKitchenPOS.Models;
using DesktopKitchenPOS.Networking.Endpoints;

namespace DesktopKitchenPOS.Services;

public class ReportService
{
    public async Task<SalesReport> GetSalesAsync(string period)
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<SalesReport>(
            HttpMethod.Get, ReportEndpoints.Sales(period));
    }

    public async Task<List<TopItemsReport>> GetTopItemsAsync(string period, int limit = 10)
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<List<TopItemsReport>>(
            HttpMethod.Get, ReportEndpoints.TopItems(period, limit));
    }

    public async Task<List<HourlyReport>> GetHourlyAsync()
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<List<HourlyReport>>(
            HttpMethod.Get, ReportEndpoints.Hourly);
    }

    public async Task<LiveDashboardData> GetLiveAsync()
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<LiveDashboardData>(
            HttpMethod.Get, ReportEndpoints.Live);
    }
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/services/ReportService.kt
package com.desktopkitchen.pos.services

import com.desktopkitchen.pos.models.HourlyReport
import com.desktopkitchen.pos.models.LiveDashboardData
import com.desktopkitchen.pos.models.SalesReport
import com.desktopkitchen.pos.models.TopItemsReport
import com.desktopkitchen.pos.networking.api.ReportApi
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ReportService @Inject constructor(
    private val reportApi: ReportApi
) {
    suspend fun getSales(period: String): SalesReport = reportApi.getSales(period)

    suspend fun getTopItems(period: String, limit: Int = 10): List<TopItemsReport> =
        reportApi.getTopItems(period, limit)

    suspend fun getHourly(): List<HourlyReport> = reportApi.getHourly()

    suspend fun getLive(): LiveDashboardData = reportApi.getLive()
}
